compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolTable.java

```diff
@@ -283,6 +284,12 @@ private SymbolTable(CompilerContext context) {
                 ImmutableTypeCloner.getImmutableIntersectionType((SelectivelyImmutableReferenceType) this.anyType,
                         this, names);
         initializeType(this.anyAndReadonly, this.anyAndReadonly.effectiveType.name.getValue(), BUILTIN);
+
+        defineReadonlyCompoundType();
+    }
+
+    private void defineReadonlyCompoundType() {
+        anyAndReadonlyOrError = BUnionType.create(null, anyAndReadonly, errorType);
```

Comment:
WDYT of inlining this method?

Reply:
I was thinking we need to move more of the code into appropriate functions to keep the constructor clean and easy to read.